api/v1/services/TokenService.js

```diff
@@ -0,0 +1,45 @@
+var Token = require('../models/Token');
+var config = require('../../config');
+var logger = require('../../logging');
+
+/**
+ * Finds a token given the Token value
+ * @param {String} value The Token's value
+ * @param {String} scope The Scope the token is for
+ * @return {Promise} resolving to the associated Token Model
+ * @throws {NotFoundError} when the requested token cannot be found
+ */
+module.exports.findTokenByValue = function(value, scope) {
+	return Token
+		.findByValue(value)
+		.then(function(result) {
+			if (!result) {
+				var message = "Could not find the provided token to reset password";
+				var source = "token";
+				throw new errors.InvalidParameterError(message, source);
```

Comment:
This is not the right error -- I think you meant `NotFoundError`. I'd also argue that the error message is way too specific, as not all callers of this method will be trying to reset a password